### ISSUE TITLE
Cannot find cast devices on Mac OS X

### ISSUE THREAD

**Issue by nathanrsm** [skorokithakis/catt#193]
Hi all. This is pretty similar to issue #98, but it's been about 6 months since the latest reply and I wondered if you might be able to assist me here. When I run the catt scan command, I receive:

```
catt scan
Scanning Chromecasts...
Error: No devices found.
```

My mac internal IP is 192.168.1.10, and the chromecast is on 192.168.1.6. I can successfully ping/lookup/throw curl requests and port scan the chromecast with no problems. I can also cast fine through the browser - it just seems to be the zeroconf side of things that isn't working! I've tried using pychromecast itself to no avail, and I've also attempted to use some python from https://github.com/jstasiak/python-zeroconf:
```python
from zeroconf import ZeroconfServiceTypes
print('\n'.join(ZeroconfServiceTypes.find()))
```
and this also returns no results when I run it. 

Has anybody else got it working on Mac or perhaps might be able to diagnose this? Appreciate it :)

**Comment by skorokithakis** [comment]
Hmm, I have a pretty weird home setup so I haven't been able to diagnose this (I'm on a different subnet so the Chromecast can't communicate anyway). If anyone could reproduce/debug this, it would be great. @nathanrsm, can you fire up Wireshark and take a look at the discovery process?

**Comment by nathanrsm** [comment]
@skorokithakis Of course! I have to admit though, I'm a bit of a newbie at this networking stuff. I've got wireshark up and running, and I filtered for the SSDP protocol. I assume you just want me to run the 'catt scan' on my mac and then capture the output from wireshark? Apologies for what is probably a basic question - and thanks for helping out! :)

**Comment by skorokithakis** [comment]
Yes, that is correct. I'm not an expert on how discovery happens either, as we just use the library, but maybe you can spot a packet not returning, or getting dropped somehow...

**Comment by skorokithakis** [comment]
I wonder if this bug should be filed at the pychromecast project, actually...

**Comment by nathanrsm** [comment]
@skorokithakis I've been looking at this over the last few hours, and from the looks of it, everything is okay. I checked that both devices were on the same wifi band (5GHz) and the SSDP looks okay to me. Interestingly though, there are two things... firstly, at the end of the SSDP messages sent from my mac (192.168.1.10) I receive a response from the ChromeCast at 192.168.1.5:

![Screenshot 2019-04-04 at 15 37 15](https://user-images.githubusercontent.com/35274081/55564323-8cbed580-56ef-11e9-8cf3-019a6cc563f5.png)

I also noticed that my mac sent an ARP request through, asking for a IPv4 link-local address. From looking at zeroconf, it looks like it uses these link-local addresses rather than the normal IP:

![Screenshot 2019-04-04 at 15 39 14](https://user-images.githubusercontent.com/35274081/55564444-d14a7100-56ef-11e9-8b5c-80316406eaf3.png)

You're right though, this might be better off over on the pychromecast project. If you could let me know if anything looks odd to you, and if not then I'll raise it over there instead. Many thanks for all the help so far :)

**Comment by skorokithakis** [comment]
Oh, I wonder if it's an IPv6 issue, I think we aren't super IPv6-compatible...

**Comment by nathanrsm** [comment]
@skorokithakis I had a chat with a networking guy at my job and he also came to a similar conclusion on the IPv6 side of things. So, I ran the ifconfig to see my mac's setup:

```ifconfig -a```

mine had something like 

```
inet6 ::1 prefixlen 128 
inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
```

Where the fe80 apparently dictates that it's using the link-local addressing. I then disabled the IPv6 on my mac with ```networksetup -setv6off "Wi-Fi"``` and then re-ran the ```ifconfig -a```... no sign of the inet6.

So, re-ran the catt scan, and would you look at that:
```
Scanning Chromecasts...
192.168.1.5 - Lounge - Google Inc. Chromecast
192.168.1.15 - Lounge Speaker - Google Inc. Google Home Mini
```

Thank you for all your help, and I hope this serves to help others who had my issue!

**Comment by skorokithakis** [comment]
That's very useful info, thanks!

**Comment by michaelmior** [comment]
Unfortunately disabling IPv6 did nothing to help me in the same situation.

**Comment by skorokithakis** [comment]
@michaelmior Are you on the same subnet as the Chromecast? Can you ping it?

**Comment by michaelmior** [comment]
Yes and yes :)

**Comment by skorokithakis** [comment]
Hm, that's very odd, I'm not sure what might be going on, but discovery is fairly straightforward on the network level, so I'm at a loss.

**Comment by michaelmior** [comment]
Interestingly, but perhaps reassuringly, it's not just pychromecast. [go-cast](https://github.com/barnybug/go-cast) has the same issue.

**Comment by skorokithakis** [comment]
Ah, on one hand, it's reassuring that it's not our fault, on the other it's unfortunate that it's probably not something we can fix...

**Comment by michaelmior** [comment]
Given that other devices (e.g. my Android phone) are able to complete discovery just fine on the same network, I'm sure there's some solution.

**Comment by rafalyesware** [comment]
For the record, I had a similar issue as described in https://github.com/skorokithakis/catt/issues/98#issuecomment-389014923 (`catt scan` worked, but attempting to play anything either just hung or gave me a `[(192.168.1.137):8009] Failed to connect, retrying in 5.0s` error).

Turns out I'd forgotten that my MacBook dock was plugged into a wired ethernet connection -- once I unplugged that, it was all good.  Of course my answer to the "is it on the same network as the Chromecast?" question was a "yes"... but I also had the wired connection on).

Just an FYI for posterity's sake.
